### PR TITLE
Add set_interval function to Timer

### LINKED ISSUE #3128: Interval of Timer can't be changed
```rust
    let timer_run = Arc::new(AtomicBool::new(false));
    let timer = Timer::default();

    let weak = window.as_weak();
    let t_r_1 = timer_run.clone();

    let timer_func = move || {
        if t_r_1.load(std::sync::atomic::Ordering::Relaxed) {
            weak.unwrap().invoke_button_step();
        }
    };
    timer.start(TimerMode::Repeated, Duration::from_millis(20), timer_func);
    let timer = Arc::new(timer);

    window.on_step_time_changed(move |interval| {
        if let Ok(interval) = interval.to_string().parse::<f32>() {
            timer.start(TimerMode::Repeated, Duration::from_secs_f32(interval), timer_func);
        }
    });
```
Code above would meet an error
```
error[E0507]: cannot move out of `timer_func`, a captured variable in an `FnMut` closure
```

Currently slint don't support to change the interval, so I must call start to change it, which leeds to nested of closure.

### PR DESCRIPTION
**Pull request by asuper0**
Add set_interval function to Timer, which could change the interval of timer without re-invoking a start function.
fix #3128

### CODE REVIEW

File: internal/core/timers.rs [hunk 1]
```diff
@@ -143,6 +143,23 @@ impl Timer {
...
+
+    /// Change the duration of timer. If the timer was previously started by calling [`Self::start()`]
+    /// with a duration and callback, then the time when the callback will be next invoked
+    /// is re-calculated to be in the specified duration relative to when this function is called.
+    ///
+    /// Does nothing if the timer was never started.
+    ///
+    /// Arguments:
+    /// * `interval`: The duration from now until when the timer should fire. And the period of that timer
+    ///    for [`Repeated`](TimerMode::Repeated) timers.
```
Comment (ogoffart) [inline internal/core/timers.rs:155]: We don't usually use this style to document the arguments.

**Review by ogoffart — COMMENTED** [review]
Thanks for the pull request!

This looks good to me. @tronical do you agree?

Please add a changelog entry.

File: internal/core/timers.rs (hunk 1 above)
Comment (tronical) [reply internal/core/timers.rs:155]: I recall trying this style once before, but at least I wasn't consistent about it. I quite like it, TBH. I don't recall where I saw it first.

**Comment by tronical** [comment]
> Thanks for the pull request!
> 
> This looks good to me. @tronical do you agree?

Yes, looks good to me, too. Thanks @asuper0

File: internal/core/timers.rs (hunk 1 above)
Comment (asuper0) [reply internal/core/timers.rs:155]: I just copy the comment from Timer::start() 😆

**Comment by ogoffart** [comment]
Thanks!

**Pull request merged**